### PR TITLE
[DatePicker] Ignore escape key when the picker is already closed

### LINKED ISSUE #4781: [DatePicker] Pressing `esc` twice anywhere on the page resets date value to a previously valid date without updating the input field
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Using the Datepicker (DesktopPicker):

- Pressing the ESC button anywhere on the page causes the onChange event to be fired, which causes erroneous dates to be set, but doesn't update the input field respectively. 
- Opening and closing the DatePicker widget causes erroneous dates to be set but doesn't update the input field respectively. 

### Expected behavior 🤔

- Pressing ESC should only close the Datepicker widget while it is open and not modify date values. 
- If Datepicker is opening and closing, it doesn't reset or modify date values unless they are explicitly chosen.

### Steps to reproduce 🕹

- If a valid date was set using the input field, pressing ESC both focused & unfocused, will reset to whatever date was previously set using the Datepicker widget or initially instantiated, including null values. But will not update the Textfield with the new values. 

- If a valid date was set using the input field,  opening the Datepicker widget and closing it, will reset the date value to the previously instantiated value, including null values. But will not update the Textfield with the new values. 

- If a valid date was set using the Datepicker widget, but modified in the input field, pressing ESC will reset to whatever was previously set. But will not update the Textfield with the new values. 

https://codesandbox.io/s/responsivedatepickers-material-demo-forked-i9xbbs?file=/demo.tsx

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Browser Used: Chrome
  System:
    OS: macOS 11.6.2
    CPU: (8) x64 Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz
    Memory: 781.23 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
    Yarn: 3.2.0 - ~/.nvm/versions/node/v14.19.0/bin/yarn
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
    Homebrew: 3.3.16 - /usr/local/bin/brew
    pip3: 21.0.1 - ~/.pyenv/shims/pip3
    RubyGems: 3.0.3 - /usr/bin/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 4.2.1 - /usr/bin/gcc
    Git: 2.35.1 - /usr/local/bin/git
    Clang: 13.0.0 - /usr/bin/clang
  Servers:
    Apache: 2.4.51 - /usr/sbin/apachectl
  Virtualization:
    Docker: 20.10.11 - /usr/local/bin/docker
    VirtualBox: 6.1.16 - /usr/local/bin/vboxmanage
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8193401
    Nano: 2.0.6 - /usr/bin/nano
    Vim: 8.2 - /usr/bin/vim
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Bash: 3.2.57 - /bin/bash
    Java: 16.0.2 - /usr/local/opt/openjdk/bin/javac
    Perl: 5.30.2 - /usr/bin/perl
    PHP: 7.3.29 - /usr/bin/php
    Python: 2.7.16 - /Users/mikewolfd/.pyenv/shims/python
    Python3: 3.9.2 - /Users/mikewolfd/.pyenv/shims/python3
    Ruby: 2.6.3 - /usr/bin/ruby
  Databases:
    SQLite: 3.32.3 - /usr/bin/sqlite3
  Browsers:
    Chrome: 98.0.4758.109
    Edge: 90.0.818.62
    Firefox: 96.0.2
    Safari: 15.3
  Monorepos:
    Yarn Workspaces: 3.2.0
    Lerna: 4.0.0
```
</details>

### PR DESCRIPTION
**Pull request by mikewolfd**
Closes #4781

### CODE REVIEW

**Review by flaviendelangle — COMMENTED** [review]
Thanks for your contribution

I transferred the issue to this repo.
Could you add a test case on `DesktopDatePicker.test.tsx` for instance ? (in the `picker state` describe, you can use the same structure as the other tests).

**Comment by mikewolfd** [comment]
@flaviendelangle Done!

File: packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePickerKeyboard.test.tsx [hunk 1]
```diff
@@ -46,6 +46,24 @@ describe('<DesktopDatePicker /> keyboard interactions', () => {
     expect(handleClose.callCount).to.equal(1);
   });
 
+  it('does not trigger onClose on Escape press if already closed', () => {
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePickerKeyboard.test.tsx:49]: Could you please put it in the `picker lifecycle` `describe` and follow the structure of the test there ("should not call onClose or onAccept when clicking outside of the picker if not opened" being a good candidate testing the exact same behavior).

That way we also test that `onAccept` and `onChange` are not called.

**Review by flaviendelangle — APPROVED** [review]

**Comment by flaviendelangle** [comment]
I let you fix the code format with `yarn prettier` and we will be able to merge it :+1:

**Pull request merged**